src/test/java/org/broadinstitute/hellbender/utils/read/SamAssertionUtils.java

```diff
@@ -17,30 +17,46 @@
  */
 public final class SamAssertionUtils {
 
-    public static void assertSamsEqual(final File sam1, final File sam2, ValidationStringency validationStringency) throws IOException {
-        try (final SamReader reader1 = SamReaderFactory.makeDefault().validationStringency(validationStringency).open(sam1);
-             final SamReader reader2 = SamReaderFactory.makeDefault().validationStringency(validationStringency).open(sam2)) {
+    private static SamReader getReader(File sam, ValidationStringency validationStringency, File reference) {
+        return SamReaderFactory.makeDefault().validationStringency(validationStringency).referenceSequence(reference).open(sam);
+    }
+
+    public static void assertSamsEqual(final File sam1, final File sam2, ValidationStringency validationStringency, final File reference) throws IOException {
```

Comment:
Add javadoc to all public methods here. For methods that take a reference, document the fact that the reference parameter IS allowed to be null, and for methods that don't take a `ValidationStringency`, document the fact that the default of `ValidationStringency.DEFAULT_STRINGENCY` will be used.